### PR TITLE
feat: support routing on Zksync

### PR DESCRIPTION
**Pull request by jsy1218**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
feat

- **What is the current behavior?** (You can also link to an open issue here)
no routing support on Zksync

- **What is the new behavior (if this is a feature change)?**
routing support on Zksync

- **Other information**:

### CODE REVIEW

**Comment by jsy1218** [comment]
* **#594** <a href="https://app.graphite.dev/github/pr/Uniswap/smart-order-router/594?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @jsy1218 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by cgkol — COMMENTED** [review]
Have we tested e2e that it works via rest calls?

**Comment by jsy1218** [comment]
> Have we tested e2e that it works via rest calls?

aw I tried local CLI https://app.warp.dev/block/XGXRjUGS8eZYEnQSJ8diVf

File: src/providers/v3/pool-provider.ts [hunk 1]
```diff
@@ -229,6 +229,8 @@ export class V3PoolProvider implements IV3PoolProvider {
       tokenA: token0,
       tokenB: token1,
       fee: feeAmount,
+      initCodeHashManualOverride: undefined,
```
Comment (mikeki) [inline src/providers/v3/pool-provider.ts:232]: what is this for? :o

**Review by mikeki — APPROVED** [review]

**Review by cgkol — APPROVED** [review]
✅

File: src/providers/v3/pool-provider.ts (hunk 1 above)
Comment (jsy1218) [reply src/providers/v3/pool-provider.ts:232]: initcode hash on ZkSync is different from other chains https://github.com/Uniswap/sdks/pull/36#discussion_r1628378049, hence we have to explicitly allow for pass in initCodeHashManualOverride for future awareness. In this case, we encapsulated the different initcode hash with v3-sdk.

**Pull request merged**